### DISCUSSION TITLE
Local SearX SSL

### DISCUSSION THREAD

**Discussion by mrtargaryen** [searx/searx#996] [Installation issue: custom]
I use HTTPS Everywhere on Firefox and block all unencrypted requests. So the local instance of Searx in Docker on my Mac can't be used in Firefox.

Is it possible to add an SSL certificate for my local Searx instance so I can use it in Firefox with HTTPS Everywhere blocking all unencrypted requests?

I have never added a SSL cert to anything before, is there a guide that I can use in order to generate and add the certificate to the Searx docker instance?

**Comment by adeweever91** [comment]
I created this image, it will run a searx instance using nginx/uwsgi and it will generate a self signed cert. Should work for HTTPS everywhere but youll have to accept the cert. 

https://hub.docker.com/r/adeweever91/searx_ss/

**Comment by mrtargaryen** [comment]
@adeweever91 Thank you so much, thats awesome. I set it up easily on local host and its working fine with HTTPS Everywhere blocking all unencrypted requests. Awesome!!

**Comment by mrtargaryen** [comment]
@adeweever91 I'm trying to edit the Searx settings.yml for Morty serch proxy but I'm unable to find it with this installation on Ubuntu. Where are your searx_ss files stored? Cheers.

**Comment by adeweever91** [comment]
Sorry for the slow reply, if you run `docker exec -it CONTAINER_NAME sh` youll get a shell inside the container, then `vi /usr/local/searx/searx/settings.yml` youll be able to edit the settings file. save the changed then exit the shell inside the container and restart the container. Do you know how to use volumes in docker? if you do just mount your settings file with `-v /path/to/your/settings.yml:/usr/local/searx/searx/settings.yml` when you run the container and then you can just make changes to the file on your system and restart the container after you have made those changes
